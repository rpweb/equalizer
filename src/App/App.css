header {
  text-align: center;
  position: relative;
}
header .logo {
  width: 90px;
  margin-top: 25px;
}
