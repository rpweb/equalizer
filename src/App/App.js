import logo from './logo.svg';
import './App.css';
import CalculatorView from '../components/CalculatorView/CalculatorView';

function App() {
  return (
    <>
      <header>
        <img src={logo} className="logo" alt="Equalizer" />
      </header>
      <main>
        <CalculatorView />
      </main>
    </>
  );
}

export default App;
