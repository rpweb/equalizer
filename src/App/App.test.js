import { render, screen, fireEvent } from '@testing-library/react';
import App from './App';

function abTest(symbol, result) {
  fireEvent.click(screen.getByText(1));
  fireEvent.click(screen.getByText(2));
  fireEvent.click(screen.getByText(2));
  fireEvent.click(screen.getByText(symbol));
  fireEvent.click(screen.getByText(1));
  fireEvent.click(screen.getByText(2));
  fireEvent.click(screen.getByAltText('='));
  expect(screen.getByText(result)).toBeInTheDocument();
  fireEvent.click(screen.getByText('C'));
}

function aTest(symbol, result) {
  fireEvent.click(screen.getByText(1));
  fireEvent.click(screen.getByText(2));
  fireEvent.click(screen.getByText(symbol));
  fireEvent.click(screen.getByAltText('='));
  expect(screen.getByText(result)).toBeInTheDocument();
  fireEvent.click(screen.getByText('C'));
}

function bracketsTest(symbol, result = 99) {
  if (symbol) {
    fireEvent.click(screen.getByText(symbol));
  }
  fireEvent.click(screen.getByText('('));
  fireEvent.click(screen.getByText(1));
  fireEvent.click(screen.getByText(1));
  fireEvent.click(screen.getByText('+'));
  fireEvent.click(screen.getByText(2));
  fireEvent.click(screen.getByText(2));
  fireEvent.click(screen.getByText(')'));
  fireEvent.click(screen.getByText('x'));
  fireEvent.click(screen.getByText(3));
  fireEvent.click(screen.getByAltText('='));
  expect(screen.getByText(result)).toBeInTheDocument();
  fireEvent.click(screen.getByText('C'));
}

function inputTest(value, result) {
  fireEvent.change(screen.getByPlaceholderText('0'), {
    target: { value },
  });
  fireEvent.keyPress(screen.getByPlaceholderText('0'), {
    charCode: '13',
  });
  expect(screen.getByText(result)).toBeInTheDocument();
  fireEvent.click(screen.getByText('C'));
}

test('a b calculations', () => {
  render(<App />);
  abTest('+', 134);
  abTest('-', 110);
  abTest('x', 1464);
  abTest('รท', 10.166666666666666);
  fireEvent.click(screen.getByText('Academic'));
  abTest('^', 1.0872213398722914e25);
});

test('a calculations', () => {
  render(<App />);
  fireEvent.click(screen.getByText('Academic'));
  aTest('!', 479001600);
  aTest('%', 0.12);
});

test('brackets calculation', () => {
  render(<App />);
  fireEvent.click(screen.getByText('Academic'));
  bracketsTest();
  bracketsTest('sqrt', 17.233687939614086);
  bracketsTest('sin', 2.9997355803218015);
  bracketsTest('cos', -0.03983024166917844);
});

test('keyboard calculation', () => {
  render(<App />);
  inputTest('500-min(sqrt(2+2*7),9,6)', '496');
  inputTest('-3!', '-6');
  inputTest('12---11+1-3', '-1');
});

test('switch calculator', () => {
  render(<App />);
  expect(screen.getByText('Academic')).toBeInTheDocument();
  fireEvent.click(screen.getByText('Academic'));
  expect(screen.getByText('sqrt')).toBeInTheDocument();
  expect(screen.getByText('Standard')).toBeInTheDocument();
  fireEvent.click(screen.getByText('Standard'));
  expect(screen.queryByText('sin')).toBeNull();
  expect(screen.getByText('Academic')).toBeInTheDocument();
});
