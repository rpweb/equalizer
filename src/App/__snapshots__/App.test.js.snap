// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot app standard test 1`] = `
Array [
  <header>
    <img
      alt="Equalizer"
      className="logo"
      src="logo.svg"
    />
  </header>,
  <main>
    <div
      className="calculator-wrapper"
    >
      <div
        className="calculator-toggle"
      >
        <button
          onClick={[Function]}
        >
          Academic
        </button>
      </div>
      <div
        className="calculator-view Standard"
      >
        <div
          className="calculator"
        >
          <input
            className=""
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="0"
            type="text"
            value=""
          />
          <div
            className="button-wrapper operation"
          >
            <button
              name="="
              onClick={[Function]}
            >
              <img
                alt="="
                name="="
                src="logo.svg"
                width={20}
              />
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="("
              onClick={[Function]}
            >
              (
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="C"
              onClick={[Function]}
            >
              C
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name=")"
              onClick={[Function]}
            >
              )
            </button>
          </div>
          <div
            className="button-wrapper operation"
          >
            <button
              name="/"
              onClick={[Function]}
            >
              รท
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="7"
              onClick={[Function]}
            >
              7
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="8"
              onClick={[Function]}
            >
              8
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="9"
              onClick={[Function]}
            >
              9
            </button>
          </div>
          <div
            className="button-wrapper operation"
          >
            <button
              name="*"
              onClick={[Function]}
            >
              x
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="4"
              onClick={[Function]}
            >
              4
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="5"
              onClick={[Function]}
            >
              5
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="6"
              onClick={[Function]}
            >
              6
            </button>
          </div>
          <div
            className="button-wrapper operation"
          >
            <button
              name="-"
              onClick={[Function]}
            >
              -
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="1"
              onClick={[Function]}
            >
              1
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="2"
              onClick={[Function]}
            >
              2
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="3"
              onClick={[Function]}
            >
              3
            </button>
          </div>
          <div
            className="button-wrapper operation"
          >
            <button
              name="+"
              onClick={[Function]}
            >
              +
            </button>
          </div>
          <div
            className="button-wrapper double"
          >
            <button
              name="0"
              onClick={[Function]}
            >
              0
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="."
              onClick={[Function]}
            >
              .
            </button>
          </div>
          <div
            className="button-wrapper operation"
          >
            <button
              name="%"
              onClick={[Function]}
            >
              %
            </button>
          </div>
        </div>
        <div
          className="formulas"
        >
          <ul />
        </div>
      </div>
    </div>
  </main>,
]
`;

exports[`snapshot app test 1`] = `
Array [
  <header>
    <img
      alt="Equalizer"
      className="logo"
      src="logo.svg"
    />
  </header>,
  <main>
    <div
      className="calculator-wrapper"
    >
      <div
        className="calculator-toggle"
      >
        <button
          onClick={[Function]}
        >
          Academic
        </button>
      </div>
      <div
        className="calculator-view Standard"
      >
        <div
          className="calculator"
        >
          <input
            className=""
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="0"
            type="text"
            value=""
          />
          <div
            className="button-wrapper operation"
          >
            <button
              name="="
              onClick={[Function]}
            >
              <img
                alt="="
                name="="
                src="logo.svg"
                width={20}
              />
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="("
              onClick={[Function]}
            >
              (
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="C"
              onClick={[Function]}
            >
              C
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name=")"
              onClick={[Function]}
            >
              )
            </button>
          </div>
          <div
            className="button-wrapper operation"
          >
            <button
              name="/"
              onClick={[Function]}
            >
              รท
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="7"
              onClick={[Function]}
            >
              7
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="8"
              onClick={[Function]}
            >
              8
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="9"
              onClick={[Function]}
            >
              9
            </button>
          </div>
          <div
            className="button-wrapper operation"
          >
            <button
              name="*"
              onClick={[Function]}
            >
              x
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="4"
              onClick={[Function]}
            >
              4
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="5"
              onClick={[Function]}
            >
              5
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="6"
              onClick={[Function]}
            >
              6
            </button>
          </div>
          <div
            className="button-wrapper operation"
          >
            <button
              name="-"
              onClick={[Function]}
            >
              -
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="1"
              onClick={[Function]}
            >
              1
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="2"
              onClick={[Function]}
            >
              2
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="3"
              onClick={[Function]}
            >
              3
            </button>
          </div>
          <div
            className="button-wrapper operation"
          >
            <button
              name="+"
              onClick={[Function]}
            >
              +
            </button>
          </div>
          <div
            className="button-wrapper double"
          >
            <button
              name="0"
              onClick={[Function]}
            >
              0
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="."
              onClick={[Function]}
            >
              .
            </button>
          </div>
          <div
            className="button-wrapper operation"
          >
            <button
              name="%"
              onClick={[Function]}
            >
              %
            </button>
          </div>
        </div>
        <div
          className="formulas"
        >
          <ul />
        </div>
      </div>
    </div>
  </main>,
]
`;

exports[`snapshot standard calcu test 1`] = `
Array [
  <header>
    <img
      alt="Equalizer"
      className="logo"
      src="logo.svg"
    />
  </header>,
  <main>
    <div
      className="calculator-wrapper"
    >
      <div
        className="calculator-toggle"
      >
        <button
          onClick={[Function]}
        >
          Academic
        </button>
      </div>
      <div
        className="calculator-view Standard"
      >
        <div
          className="calculator"
        >
          <input
            className=""
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="0"
            type="text"
            value=""
          />
          <div
            className="button-wrapper operation"
          >
            <button
              name="="
              onClick={[Function]}
            >
              <img
                alt="="
                name="="
                src="logo.svg"
                width={20}
              />
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="("
              onClick={[Function]}
            >
              (
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="C"
              onClick={[Function]}
            >
              C
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name=")"
              onClick={[Function]}
            >
              )
            </button>
          </div>
          <div
            className="button-wrapper operation"
          >
            <button
              name="/"
              onClick={[Function]}
            >
              รท
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="7"
              onClick={[Function]}
            >
              7
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="8"
              onClick={[Function]}
            >
              8
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="9"
              onClick={[Function]}
            >
              9
            </button>
          </div>
          <div
            className="button-wrapper operation"
          >
            <button
              name="*"
              onClick={[Function]}
            >
              x
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="4"
              onClick={[Function]}
            >
              4
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="5"
              onClick={[Function]}
            >
              5
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="6"
              onClick={[Function]}
            >
              6
            </button>
          </div>
          <div
            className="button-wrapper operation"
          >
            <button
              name="-"
              onClick={[Function]}
            >
              -
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="1"
              onClick={[Function]}
            >
              1
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="2"
              onClick={[Function]}
            >
              2
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="3"
              onClick={[Function]}
            >
              3
            </button>
          </div>
          <div
            className="button-wrapper operation"
          >
            <button
              name="+"
              onClick={[Function]}
            >
              +
            </button>
          </div>
          <div
            className="button-wrapper double"
          >
            <button
              name="0"
              onClick={[Function]}
            >
              0
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="."
              onClick={[Function]}
            >
              .
            </button>
          </div>
          <div
            className="button-wrapper operation"
          >
            <button
              name="%"
              onClick={[Function]}
            >
              %
            </button>
          </div>
        </div>
        <div
          className="formulas"
        >
          <ul />
        </div>
      </div>
    </div>
  </main>,
]
`;

exports[`snapshot standard calculator test 1`] = `
Array [
  <header>
    <img
      alt="Equalizer"
      className="logo"
      src="logo.svg"
    />
  </header>,
  <main>
    <div
      className="calculator-wrapper"
    >
      <div
        className="calculator-toggle"
      >
        <button
          onClick={[Function]}
        >
          Academic
        </button>
      </div>
      <div
        className="calculator-view Standard"
      >
        <div
          className="calculator"
        >
          <input
            className=""
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="0"
            type="text"
            value=""
          />
          <div
            className="button-wrapper operation"
          >
            <button
              name="="
              onClick={[Function]}
            >
              <img
                alt="="
                name="="
                src="logo.svg"
                width={20}
              />
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="("
              onClick={[Function]}
            >
              (
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="C"
              onClick={[Function]}
            >
              C
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name=")"
              onClick={[Function]}
            >
              )
            </button>
          </div>
          <div
            className="button-wrapper operation"
          >
            <button
              name="/"
              onClick={[Function]}
            >
              รท
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="7"
              onClick={[Function]}
            >
              7
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="8"
              onClick={[Function]}
            >
              8
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="9"
              onClick={[Function]}
            >
              9
            </button>
          </div>
          <div
            className="button-wrapper operation"
          >
            <button
              name="*"
              onClick={[Function]}
            >
              x
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="4"
              onClick={[Function]}
            >
              4
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="5"
              onClick={[Function]}
            >
              5
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="6"
              onClick={[Function]}
            >
              6
            </button>
          </div>
          <div
            className="button-wrapper operation"
          >
            <button
              name="-"
              onClick={[Function]}
            >
              -
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="1"
              onClick={[Function]}
            >
              1
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="2"
              onClick={[Function]}
            >
              2
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="3"
              onClick={[Function]}
            >
              3
            </button>
          </div>
          <div
            className="button-wrapper operation"
          >
            <button
              name="+"
              onClick={[Function]}
            >
              +
            </button>
          </div>
          <div
            className="button-wrapper double"
          >
            <button
              name="0"
              onClick={[Function]}
            >
              0
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="."
              onClick={[Function]}
            >
              .
            </button>
          </div>
          <div
            className="button-wrapper operation"
          >
            <button
              name="%"
              onClick={[Function]}
            >
              %
            </button>
          </div>
        </div>
        <div
          className="formulas"
        >
          <ul />
        </div>
      </div>
    </div>
  </main>,
]
`;

exports[`snapshot standard test 1`] = `
Array [
  <header>
    <img
      alt="Equalizer"
      className="logo"
      src="logo.svg"
    />
  </header>,
  <main>
    <div
      className="calculator-wrapper"
    >
      <div
        className="calculator-toggle"
      >
        <button
          onClick={[Function]}
        >
          Academic
        </button>
      </div>
      <div
        className="calculator-view Standard"
      >
        <div
          className="calculator"
        >
          <input
            className=""
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="0"
            type="text"
            value=""
          />
          <div
            className="button-wrapper operation"
          >
            <button
              name="="
              onClick={[Function]}
            >
              <img
                alt="="
                name="="
                src="logo.svg"
                width={20}
              />
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="("
              onClick={[Function]}
            >
              (
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="C"
              onClick={[Function]}
            >
              C
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name=")"
              onClick={[Function]}
            >
              )
            </button>
          </div>
          <div
            className="button-wrapper operation"
          >
            <button
              name="/"
              onClick={[Function]}
            >
              รท
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="7"
              onClick={[Function]}
            >
              7
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="8"
              onClick={[Function]}
            >
              8
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="9"
              onClick={[Function]}
            >
              9
            </button>
          </div>
          <div
            className="button-wrapper operation"
          >
            <button
              name="*"
              onClick={[Function]}
            >
              x
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="4"
              onClick={[Function]}
            >
              4
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="5"
              onClick={[Function]}
            >
              5
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="6"
              onClick={[Function]}
            >
              6
            </button>
          </div>
          <div
            className="button-wrapper operation"
          >
            <button
              name="-"
              onClick={[Function]}
            >
              -
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="1"
              onClick={[Function]}
            >
              1
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="2"
              onClick={[Function]}
            >
              2
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="3"
              onClick={[Function]}
            >
              3
            </button>
          </div>
          <div
            className="button-wrapper operation"
          >
            <button
              name="+"
              onClick={[Function]}
            >
              +
            </button>
          </div>
          <div
            className="button-wrapper double"
          >
            <button
              name="0"
              onClick={[Function]}
            >
              0
            </button>
          </div>
          <div
            className="button-wrapper "
          >
            <button
              name="."
              onClick={[Function]}
            >
              .
            </button>
          </div>
          <div
            className="button-wrapper operation"
          >
            <button
              name="%"
              onClick={[Function]}
            >
              %
            </button>
          </div>
        </div>
        <div
          className="formulas"
        >
          <ul />
        </div>
      </div>
    </div>
  </main>,
]
`;
